### PR TITLE
microprofile-health-81 Ability to execute only some health checks

### LINKED ISSUE #81: Ability to execute only some health checks
It would be useful in the wire format if we specified a way to execute only some health checks, rather than all discovered `@Health` health checks.

I could see this being done from both client and server side.  From client perspective, listing out a set of names to run, or a group name to run.  On the server, I could see marker interfaces, annotations, or even a simple use of class name/`@Inject @Named` support.

### PR DESCRIPTION
**Pull request by antoinesd**
fixes #81 
Adding annotations
Amend specification
Add TCK tests

Signed-off-by: Antoine Sabot-Durand <antoine@sabot-durand.net>

### CODE REVIEW

File: tck/src/main/java/org/eclipse/microprofile/health/tck/MultipleCustomFailedTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+public class MultipleCustomFailedTest extends TCKBase {
+
+    @Deployment
+    public static Archive getDeployment() {
+        return createWarFileWithClasses(MultipleCustomFailedTest.class.getSimpleName(),
+                                        FailedCustom.class, SuccessfulCustom.class);
+    }
+
+    /**
+     * Verifies the liveness health integration with CDI at the scope of a server runtime
```
Comment (xstefank) [inline tck/src/main/java/org/eclipse/microprofile/health/tck/MultipleCustomFailedTest.java]: probably just oversee from test copy, but it should state custom health checks not liveness health integration

File: tck/src/main/java/org/eclipse/microprofile/health/tck/MultipleCustomFailedTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+     */
+    @Test
+    @RunAsClient
+    public void testFailureLivenessResponsePayload() {
```
Comment (xstefank) [inline tck/src/main/java/org/eclipse/microprofile/health/tck/MultipleCustomFailedTest.java]: same as above

File: tck/src/main/java/org/eclipse/microprofile/health/tck/MultipleCustomFailedTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        Response response = getUrlCustomHealthContents("goup2");
```
Comment (xstefank) [inline tck/src/main/java/org/eclipse/microprofile/health/tck/MultipleCustomFailedTest.java]: typo "goup2"

File: tck/src/main/java/org/eclipse/microprofile/health/tck/SingleCustomSuccessfulTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        return createWarFileWithClasses(SingleCustomSuccessfulTest.class.getSimpleName(), SuccessfulCustom.class);
+    }
+
+    /**
+     * Verifies the health integration with CDI at the scope of a server runtime
+     */
+    @Test
+    @RunAsClient
+    public void testSuccessResponsePayload() {
+        Response response = getUrlCustomHealthContents("goup1");
```
Comment (xstefank) [inline tck/src/main/java/org/eclipse/microprofile/health/tck/SingleCustomSuccessfulTest.java]: typo goup1

File: tck/src/main/java/org/eclipse/microprofile/health/tck/deployment/FailedCustom.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthGroup;
+
+/**
+ * @author Antone Sabot-Durand
+ * @since 2.0
+ */
+@HealthGroup("group2")
```
Comment (xstefank) [inline tck/src/main/java/org/eclipse/microprofile/health/tck/deployment/FailedCustom.java]: so the tests probably won't pass with the above typos :)

**Comment by xstefank** [comment]
@antoinesd did you verify that the new TCK can pass?

**Comment by antoinesd** [comment]
Not yet. Do you want implement the feature in smallRye to test or do you prefer that I do it ?

**Comment by xstefank** [comment]
@antoinesd I implemented this in SmallRye. Everything passes. Thanks.

File: spec/src/main/asciidoc/java-api.adoc [hunk 6]
```diff
@@ -60,6 +61,18 @@ This means that if this procedure fails the application can be discarded (termin
 
 The `@Liveness` annotation must be applied on a `HealthCheck` implementation to define a Liveness check procedure, otherwise, this annotation is ignored.
 
+=== Custom check
+
+A new kind of Health Check can be defined with the `@HealthGroup` annotation.
+
+One or more Health Check procedure(s) can be annotated with `@HealthGroup` to apply a custom group name to the procedure.
+
+All the Health Check procedures belonging to the same group defined a new kind of Health Check.
```
Comment (xstefank) [inline spec/src/main/asciidoc/java-api.adoc]:
```suggestion
All the Health Check procedures belonging to the same group define a new kind of Health Check.
```

File: spec/src/main/asciidoc/java-api.adoc [hunk 7]
```diff
@@ -60,6 +61,18 @@ This means that if this procedure fails the application can be discarded (termin
...
+
+The `@HealthGroup` annotation must be applied on a `HealthCheck` implementation to define a custom check procedure with the given name, otherwise, this annotation is ignored.
+
+An `HealthCheck` implementation can have multiple `@HealthGroup` annotation to make it member of multiple custom Health Check's group.
```
Comment (xstefank) [inline spec/src/main/asciidoc/java-api.adoc]:
```suggestion
A `HealthCheck` implementation can have multiple `@HealthGroup` annotations to make it a member of multiple custom Health Check's groups.
```

**Comment by xstefank** [comment]
@antoinesd thanks for the updates. Should we merge this PR and add Literal in a separate one? Or are you still going to include it here?

**Comment by antoinesd** [comment]
I'll create anther PR for literals since it could be useful to have ones for `@Liveness` `@Readiness` and `@Health`

**Pull request merged**